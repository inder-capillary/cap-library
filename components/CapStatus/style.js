import styled from "styled-components";

export const StatusDot = styled.div`
  width: ${(props) => props.width};
  height: ${(props) => props.height};
  background-color: ${(props) => props.color};
  border-radius: 50%;
  display: inline-block;
  margin-right: 8px;
`;
