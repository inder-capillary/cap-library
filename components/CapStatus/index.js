import React from "react";
import PropTypes from 'prop-types';
import * as styledVars from "../styled/variables";
import CapRow from "../CapRow";
import CapLabel from "../CapLabel";
import { StatusDot } from "./style";

const { CAP_PRIMARY, CAP_SPACE_08, CAP_BLUE02, CAP_ORANGE02, CAP_RED03 } = styledVars;

const CapStatus = (props) => {
  const { type, color, text, height, width } = props;

  const typeColorMapping = {
    draft: CAP_BLUE02,
    awaitingApproval: CAP_ORANGE02,
    approved: CAP_PRIMARY.base,
    rejected: CAP_RED03,
  };

  return (
    <CapRow type="flex" align="middle">
      <StatusDot color={typeColorMapping[type] || color} height={height} width={width}></StatusDot>
      <CapLabel type="label2">{text}</CapLabel>
    </CapRow>
  );
};

CapStatus.defaultProps = {
  width: CAP_SPACE_08,
  height: CAP_SPACE_08,
  type: "",
  color: "",
  text: "",
};

CapStatus.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  color: PropTypes.string,
  type: PropTypes.string,
  text: PropTypes.string,
};

export default CapStatus;
