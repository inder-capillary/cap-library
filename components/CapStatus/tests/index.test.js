import React from "react";
import { shallow } from "enzyme";
import CapStatus from "../index";

const MountComponent = () => shallow(<CapStatus type="approved" text="Approved" />);

describe("CapStatus", () => {
  it("Should render the component", () => {
    const wrapper = MountComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
