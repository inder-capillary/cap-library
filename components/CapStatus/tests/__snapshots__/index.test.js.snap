// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapStatus Should render the component 1`] = `
<CapRow
  align="middle"
  type="flex"
>
  <styled.div
    color="#47af46"
    height="0.571rem"
    width="0.571rem"
  />
  <styled.div
    type="label2"
  >
    Approved
  </styled.div>
</CapRow>
`;
